src/App.js

```diff
@@ -1,20 +1,32 @@
-import Header from './components/Header/Header';
-import Slideshow from './components/Slideshow/Slideshow';
-import Categories from './components/Categories/Categories';
-import Products from './components/Products/Products';
-import Footer from './components/Footer/Footer';
-import './App.css';
+import { useState } from "react";
+import Header from "./components/Header/Header";
+import Slideshow from "./components/Slideshow/Slideshow";
+import Categories from "./components/Categories/Categories";
+import Products from "./components/Products/Products";
+import { Button } from "./components/Products/Products.styled";
+import ProductList from "./components/ProductList/ProductList";
+import Footer from "./components/Footer/Footer";
+import "./App.css";
+
+export default function App() {
+  const [activeProductList, setActiveProductList] = useState(false);
 
-function App() {
   return (
     <div className="App">
-      <Header/>
-      <Slideshow/>
-      <Categories/>
-      <Products/>
-      <Footer/>
-    </div>
-  );
-}
+      <Header setActiveProductList={setActiveProductList}/>
+      {activeProductList ? (
+        <ProductList />
+      ) : (
+        <>
+          <Slideshow/>
+          <Categories/>
+          <Products/>
+        </>
+      )}
+      <Button 
+      className="btn-container" 
+      onClick={e => setActiveProductList(!activeProductList)} >View all products</Button>
```

Comment:
in this case the event (e) i think is not required, the function might works also by this way:

()=>setActiveProductList(!activeProducListt)

can we check it? i'm not 100% sure